pkg/kubelet/network/plugins.go

```diff
@@ -47,6 +48,22 @@ type NetworkPlugin interface {
 
 	// TearDownPod is the method called before a pod's infra container will be deleted
 	TearDownPod(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) error
+
+	// Status is the method called to obtain the ipv4 and pv6 addresses of the container
+	Status(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) (*NetworkStatus, error)
+}
+
+type InterfaceInfo struct {
+	Iface   net.Interface `json:"iFace" description:"A network interface on the pod"`
+	IPAddrs []net.IP      `json:"ipAddrs" description:"List of IP Addresses associated with the given interface"`
+}
+
+// Stores the network status of a pod (currently just the network addresses and interfaces)
+type NetworkStatus struct {
+	// Primary ipv4/ipv6 address of the pod
+	IPAddress net.IP `json:"ipAddress" description:"Primary address of the pod"`
+	// List of network interfaces that may be optionally provided
```

Comment:
If you track the discussion early in the PR, there have been suggestions from the community to extend the information returned. Today, the PodStatus only stores one IP address, but likely that network plugins will have multiple addresses and interfaces (libnetwork for example). Also helpful for the case for IPv6.

So, while the 'exec' plugin flow does not really use the extra bits, there is scope for a plugin to want to manage more. Maybe through annotations.

I would invite @jainvipin to comment about this.